File: backend/repository/stock_item.go
package repository

import (
	"context"
	"time"

	"github.com/google/uuid"
)

type StockItem struct {
	ID        uuid.UUID `json:"id" db:"id"`
	Name      string    `json:"name" db:"name"`
	Category  string    `json:"category" db:"category"`
	ImageURL  *string   `json:"imageUrl" db:"image_url"`
	WantToBuy bool      `json:"wantToBuy" db:"want_to_buy"`
	CreatedAt time.Time `json:"createdAt" db:"created_at"`
	UpdatedAt time.Time `json:"updatedAt" db:"updated_at"`
}

type StockItemRepository interface {
	List(ctx context.Context) ([]StockItem, error)
	Create(ctx context.Context, name, category string) (*StockItem, error)
	Update(ctx context.Context, id uuid.UUID, params UpdateParams) (*StockItem, error)
	Delete(ctx context.Context, id uuid.UUID) error
}

type UpdateParams struct {
	Name      *string
	Category  *string
	WantToBuy *bool
}

File: frontend/src/types/stockItem.ts
type StockItem = {
  id: string;
  name: string;
  category: string;
  imageUrl: string | null;
  sourceUrl: string | null;
  wantToBuy: boolean;
  createdAt: string;
  updatedAt: string;
  sortedAt: string;
};

type CreateStockItemRequest = {
  name: string;
  category: string;
  wantToBuy?: boolean;
  sourceUrl?: string;
};

type UpdateStockItemRequest = {
  name?: string;
  category?: string;
  wantToBuy?: boolean;
  imageUrl?: string | null;
  sourceUrl?: string | null;
};

type ImageSearchResult = {
  imageUrl: string;
  thumbnailUrl: string;
  title: string;
};

type ErrorResponse = {
  message: string;
};

export type {
  CreateStockItemRequest,
  ErrorResponse,
  ImageSearchResult,
  StockItem,
  UpdateStockItemRequest,
};
